src/clients/UBAClient/UBAClientWithRefresh.ts

```diff
@@ -182,4 +163,55 @@ export class UBAClient {
 
     return { valid: true };
   }
+
+  protected async instantiateUBAFeeCalculator(chainId: number, token: string, fromBlock: number): Promise<void> {
+    const config = this.hubPoolClient.configStoreClient;
+    if (!isDefined(this.spokeUBAFeeCalculators[chainId]?.[token])) {
+      const spokeFeeCalculator = new UBAFeeSpokeCalculator(
+        chainId,
+        token,
+        this.getFlows(chainId, fromBlock),
+        0,
+        await config.getUBAFeeConfig(chainId, token)
+      );
+      this.spokeUBAFeeCalculators[chainId] = this.spokeUBAFeeCalculators[chainId] ?? {};
+      this.spokeUBAFeeCalculators[chainId][token] = spokeFeeCalculator;
+    }
+  }
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  protected async computeRealizedLpFee(
+    l1TokenAddress: string,
+    depositChainId: number,
+    refundChainId: number,
+    amount: BigNumber
+  ): Promise<BigNumber> {
+    const ubaConfig = await this.hubPoolClient.configStoreClient.getUBAFeeConfig(depositChainId, l1TokenAddress);
+    return computeRealizedLpFeeForRefresh(
+      l1TokenAddress,
+      depositChainId,
+      refundChainId,
+      amount,
+      this.hubPoolClient,
+      this.spokePoolClients,
+      ubaConfig.getBaselineFee(refundChainId, depositChainId),
+      ubaConfig.getLpGammaFunctionTuples(depositChainId)
+    );
+  }
+
+  protected async computeRelayerFees(
+    tokenSymbol: string,
+    amount: BigNumber,
+    depositChainId: number,
+    refundChainId: number,
+    tokenPrice?: number
+  ): Promise<RelayerFeeDetails> {
+    const relayCalculator = new RelayFeeCalculator(this.relayerConfiguration);
```

Comment:
Out of curiosity, is there a reason for the RelayFeeCalculator class to be instantiated each time this method is called? I expect it can be called quite a bit by the relayer, so it seems somewhat expensive to be treating it as a disposable object.